spec/features/consumer/shopping/embedded_shopfronts_spec.rb

```diff
@@ -115,13 +115,15 @@ def wait_for_shop_loaded
   end
 
   def login_with_modal
-    expect(page).to have_selector 'div.login-modal', visible: true
+    page.has_selector? 'div.login-modal', visible: true
 
     within 'div.login-modal' do
       fill_in "Email", with: user.email
       fill_in "Password", with: user.password
       find('input[type="submit"]').click
     end
+
+    page.has_no_selector? 'div.login-modal', visible: true
```

Comment:
This check to see if the modal was closed before moving on is what we needed, I've moved it into the `login_with_modal` method instead of putting it in the test.